### PR TITLE
added dto, mapstruct and mapper

### PR DESCRIPTION
**Pull request by ohatsenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -84,9 +105,14 @@
...
             </plugin>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+            </plugin>
         </plugins>
     </build>
 
-</project>
+</project>
```
Comment (DmytroV95) [inline pom.xml]: No empty line

File: src/main/java/com/example/onlinebookstore/repository/BookRepository.java [hunk 2]
```diff
@@ -7,4 +7,6 @@ public interface BookRepository {
     Book save(Book book);
 
     List<Book> findAll();
+
+    Book getById(Long id);
```
Comment (DmytroV95) [inline src/main/java/com/example/onlinebookstore/repository/BookRepository.java]: I think we need to use Optional return type to be sure we will not get NPE if the necessary book is not found.

**Review by DmytroV95 — CHANGES_REQUESTED** [review]
GJ! Let's improve it!

File: pom.xml [hunk 3]
```diff
@@ -24,46 +27,64 @@
...
-            <artifactId>mysql-connector-java</artifactId>
-            <version>8.0.32</version>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
         </dependency>
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
-            <optional>true</optional>
+            <version>1.18.28</version>
```
Comment (vovasalyha) [inline pom.xml:44]: lombok version is provided by parent pom

File: src/main/java/com/example/onlinebookstore/repository/BookRepository.java [hunk 4]
```diff
@@ -2,9 +2,12 @@
...
 import com.example.onlinebookstore.model.Book;
 import java.util.List;
+import java.util.Optional;
 
 public interface BookRepository {
     Book save(Book book);
 
     List<Book> findAll();
+
+    Optional<Book> getById(Long id);
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/repository/BookRepository.java:12]: in repositories it's common to name methods `find`
```suggestion
    Optional<Book> findById(Long id);
```

**Review by vovasalyha — APPROVED** [review]
GJ, take a look on few minor suggestions.

**Pull request merged**